sympy/polys/tests/test_polytools.py

```diff
@@ -1220,6 +1221,30 @@ def test_Poly_total_degree():
     assert total_degree(Poly(y**2 + x**3 + z**4, x), z) == 4
     assert total_degree(Poly(x**9 + x*z*y + x**3*z**2 + z**7,x), z) == 7
 
+def test_Poly_power_list():
+    assert Poly(0, x).power_list() == [(0,)]
+    assert Poly(0, x, y).power_list() == [(0,), (0,)]
+    assert Poly(0, x, y, z).power_list() == [(0,), (0,), (0,)]
+
+    assert Poly(1, x).power_list() == [(0,)]
+    assert Poly(1, x, y).power_list() == [(0,), (0,)]
+    assert Poly(1, x, y, z).power_list() == [(0,), (0,), (0,)]
+
+    assert Poly(x**2*y + x**3*z**2 + 1).power_list() == [(3, 2, 0), (1, 0), (2, 0)]
+
+    assert power_list(x, x) == [(1,)]
+
+    assert power_list(x*y**2 + x + 1) == [(1, 0), (2, 0)]
+    assert power_list(x*y**2 + x) == [(1,), (2, 0)] # power of y is zero with x
+    assert power_list(x*y**2) == [(1,), (2,)]
+
+    assert power_list(x**8*(1 - x)**9) == [(17, 16, 15, 14, 13, 12, 11, 10, 9, 8)]
+    # >>>expand(x**8*(1 - x)**9)
```

Comment:
A good place for this is in the commit message. Once verified, it is no longer necessary.

Reply:
Okay will be removed in next commit.